internal/proxy/default_limit_reducer.go

```diff
@@ -39,42 +38,38 @@ func (r *defaultLimitReducer) afterReduce(result *milvuspb.QueryResults) error {
 	outputFieldsID := r.req.GetOutputFieldsId()
 
 	result.CollectionName = collectionName
+	var err error
 
-	if len(result.FieldsData) > 0 {
-		result.Status = merr.Status(nil)
-	} else {
-		result.Status = &commonpb.Status{
-			ErrorCode: commonpb.ErrorCode_EmptyCollection,
-			Reason:    "empty collection", // TODO
-		}
-		return nil
-	}
-
-	for i := 0; i < len(result.FieldsData); i++ {
+	for i := 0; i < len(result.GetFieldsData()); i++ {
+		// drop ts column
 		if outputFieldsID[i] == common.TimeStampField {
 			result.FieldsData = append(result.FieldsData[:i], result.FieldsData[(i+1):]...)
+			outputFieldsID = append(outputFieldsID[:i], outputFieldsID[i+1:]...)
 			i--
 			continue
 		}
-		for _, field := range schema.Fields {
-			if field.FieldID == outputFieldsID[i] {
-				// deal with the situation that offset equal to or greater than the number of entities
-				if result.FieldsData[i] == nil {
-					var err error
-					result.FieldsData[i], err = typeutil.GenEmptyFieldData(field)
-					if err != nil {
-						return err
-					}
-				}
-				result.FieldsData[i].FieldName = field.Name
-				result.FieldsData[i].FieldId = field.FieldID
-				result.FieldsData[i].Type = field.DataType
-				result.FieldsData[i].IsDynamic = field.IsDynamic
+		field := typeutil.GetField(schema, outputFieldsID[i])
+		if field == nil {
+			err = merr.WrapErrFieldNotFound(outputFieldsID[i])
+			break
+		}
+
+		if result.FieldsData[i] == nil {
+			result.FieldsData[i], err = typeutil.GenEmptyFieldData(field)
+			if err != nil {
+				return err
```

Comment:
this branch return error directly without setting status like `field is nil` branch